src/home_robot_hw/home_robot_hw/env/stretch_abstract_env.py

```diff
@@ -533,18 +550,30 @@ def navigate_to(
         if relative:
             xyt_base = sophus2xyt(self._t_base_odom)
             xyt_goal = xyt_base_to_global(xyt, xyt_base)
-            print("Sending relative goal:")
-            print("base =", xyt_base)
-            print("goal =", xyt_goal)
         else:
             xyt_goal = xyt
 
+        # Clear self.at_goal
+        self._at_goal = False
+        self._goal_reset_t = rospy.Time.now()
+
         # Set goal
         goal_matrix = xyt2sophus(xyt_goal).matrix()
         self.goal_visualizer(goal_matrix)
         msg = matrix_to_pose_msg(goal_matrix)
         self._goal_pub.publish(msg)
 
+        if blocking:
+            rospy.sleep(self.msg_delay_t)
+            rate = rospy.Rate(10)
+            while not rospy.is_shutdown():
+                if self._at_goal:
+                    break
+                else:
+                    rate.sleep()
+            # TODO: this should be unnecessary
+            rospy.sleep(self.msg_delay_t * 10)
```

Comment:
What is this for?
Can we remove it if it's unnecessary and add more comments if it is for now?

Reply:
its a hack to sync depth and pose - i'm going to add a comment and hide it for now